lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -53,22 +52,62 @@
 
 logger = logging.getLogger(__name__)
 
-# The unique Charmhub library identifier, never change it
-LIBID = "c0f1d8f94bdd41a781fe2871e1922480"
-
-# Increment this major API version when introducing breaking changes
-LIBAPI = 0
 
-# Increment this PATCH version before using `charmcraft publish-lib` or reset
-# to 0 if you are raising the major API version
-LIBPATCH = 1
+class ExtraUserRolePermissions(Enum):
+    """An enum of user types and their associated permissions."""
+
+    # Default user has CRUD in a specific index. Update index_patterns to include the index to
+    # which these permissions are applied.
+    DEFAULT = {
+        "cluster_permissions": ["cluster_monitor"],
+        "index_permissions": [
+            {
+                "index_patterns": [],
+                "dls": "",
+                "fls": [],
+                "masked_fields": [],
+                "allowed_actions": [
+                    "indices_monitor",
+                    "crud",
+                    "data_access",  # How is this different to crud?
```

Comment:
Only `data_access` should stay:
> Combines the `crud` action group with indices:data/*.

For [reference](https://opensearch.org/docs/latest/security/access-control/default-action-groups/)